helpers/deploymentConfig.ts

```diff
@@ -12,14 +11,22 @@ export enum InterestRateModels {
   JumpRate,
 }
 
-export type TokenConfig = {
-  isMock: boolean;
-  name?: string;
-  symbol: string;
-  decimals?: number;
-  tokenAddress: string;
-  faucetInitialLiquidity?: boolean;
-};
+export type TokenConfig =
+  | {
+      isMock: true;
+      name?: string;
+      symbol: string;
+      decimals?: number;
+      faucetInitialLiquidity?: boolean;
+    }
+  | {
+      isMock: false;
+      name?: string;
+      symbol: string;
+      decimals?: number;
+      tokenAddress: string;
+      faucetInitialLiquidity?: boolean;
+    };
```

Comment:
Not sure about formatting but maybe extracting the common fields would make the intent more clear?

```suggestion
  (
    {
      isMock: true;
    }
    | {
      isMock: false;
      tokenAddress: string;
    }
  )
  & {
    name?: string;
    symbol: string;
    decimals?: number;
    faucetInitialLiquidity?: boolean;
  }
```

Reply:
clarified